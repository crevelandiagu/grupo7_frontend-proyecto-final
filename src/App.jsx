import { Typography, Button } from "@mui/material"
function App() {

  return (
    <>
      <Typography variant="h1">ABC Jobs</Typography>
      <Button variant="contained">Login</Button>
    </>
  )
}

export default App
