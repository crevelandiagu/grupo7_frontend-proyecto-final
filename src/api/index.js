export { default as candidateApi } from './candidateApi';
export { default as companyApi } from './companyApi';
export { default as companyEmployeesApi } from './companyEmployeesApi';
export { default as projectsApi } from './projectsApi';
export { default as searchToolApi } from './searchToolApi';
export { default as selectionProcess } from './selectionProcess';
export { default as performance } from './performanceApi';
