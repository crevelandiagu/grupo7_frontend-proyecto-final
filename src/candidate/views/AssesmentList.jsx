import { Grid, Typography } from '@mui/material';
import Box from '@mui/material/Box';
import { DataGrid } from '@mui/x-data-grid';
import { getEnvSelectionProcess } from '../../helpers/getEnvVaribles';
import { useAuthStore, useFetch } from '../../hooks';

const columns = [
  { field: 'id', headerName: 'ID', width: 130 },
  { field: 'company', headerName: 'Company', width: 130 },
  {
    field: 'score',
    headerName: 'Score',
    type: 'number',
    width: 80,
    editable: true,
  },
  {
    field: 'date',
    headerName: 'Date',
    width: 130,
    editable: true,
  },
  {
    field: 'time',
    headerName: 'Time',
    type: 'time',
    width: 80,
    editable: true,
  },
];

const selectionProcess = getEnvSelectionProcess();

export const AssesmentList = () => {
  const { id } = useAuthStore();
  const { data, loading } = useFetch(`${selectionProcess}/interviews/candidate/${id}`)

  console.log(loading, !data,  loading && !data)

  return (
    <Box>
      <Grid container justifyContent="center" mt={3} mb={3} >
        <Typography component="h1" variant="h4">
          List Interviews
        </Typography>
      </Grid>
      <DataGrid 
        columns={columns}
        rows={  
            data?.map((item, index) => ({
            id: index+1,
            company: item.company_id,
            score: item.score || 'Pending',
            date: item.date_interview.split('T')[0],
            time: item.date_interview.split('T')[1]
          })) || []
        }
        initialState={{
          pagination: {
            paginationModel: {
              pageSize: 5,
            },
          },
        }}
        pageSizeOptions={[5]}
        checkboxSelection
        disableRowSelectionOnClick
      />
    </Box >
  );
}