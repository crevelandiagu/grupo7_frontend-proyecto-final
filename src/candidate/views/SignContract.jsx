import Button from '@mui/material/Button';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import Container from '@mui/material/Container';

import { useAuthStore, useFetch } from '../../hooks';
import { Alert, TextField } from '@mui/material';
import contractApi  from '../../api/contractApi';
import { getEnvContract } from '../../helpers/getEnvVaribles';


const signContract = async (candidateId) => {
  try {
    const { data } = await contractApi.post(`/candidate/${candidateId}`)
    console.log('data', data);
    return data.message;
  } catch (error) {
    console.log('error', error);
  }
}

const contract = getEnvContract();

export const SignContract = () => {

  const { id, errorMessage } = useAuthStore();
  const { data } = useFetch(`${contract}/candidate/${id}`);

  const handleSubmit = (event) => {
    event.preventDefault();
    signContract(id);
  };

  return (
    <Container component="main" maxWidth="xs">
      <Box
        sx={{
          marginTop: 8,
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'center',
        }}
      >
        <Typography component="h1" variant="h4">
          Sign Contract
        </Typography>
        <Box component="form" onSubmit={handleSubmit} sx={{ mt: 3 }}>
          <TextField
            id="contract"
            disabled
            multiline
            fullWidth
            rows={10}
            readOnly
            value={ data?.message || data?.contract }
          />

          <Grid item sx={{ mt: 2 }}
            xs={12}
            display={errorMessage ? '' : 'none'}
          >
            <Alert severity="error">{errorMessage}</Alert>
          </Grid>
          <Button
            type="submit"
            disabled = {data?.message === "We don't have contracts yet" ? true : false}
            fullWidth
            variant="contained"
            sx={{ mt: 3, mb: 2 }}
          >
            Sign contract
          </Button>
        </Box>
      </Box>
    </Container>
  );
}
