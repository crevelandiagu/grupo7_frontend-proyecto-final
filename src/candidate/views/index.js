export * from './CandidateDashboard';
export * from './NothingSelectedView';