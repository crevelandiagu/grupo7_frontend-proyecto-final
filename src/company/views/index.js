export * from './CompanyDashboard';
export * from './CreateEmployeeAccount';
export * from './CreateInterview';
export * from './Project';
export * from './SearchCandidate';
export * from './DetailProfile';
export * from './AssignProject';
export * from './Interview';
export * from './AssesmentList';
export * from './Assesment';
export * from './ContractList';
export * from './PerformanceList';
export * from './Performance';
