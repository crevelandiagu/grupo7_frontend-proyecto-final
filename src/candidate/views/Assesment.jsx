import Button from '@mui/material/Button';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import Container from '@mui/material/Container';

import { useAuthStore, useFetch } from '../../hooks';
import { Alert, FormControl, FormControlLabel, FormLabel, Radio, RadioGroup } from '@mui/material';
import { candidateApi, selectionProcess } from '../../api';


const sendAssesment = async (id = 1,) => {
  try {
    const { data } = await candidateApi.get(`/assement/take-exam/${id}/candidate`)
    console.log('data', data);
    return data.message;
  } catch (error) {
    console.log('error', error);
  }
}

const questions = [
  {
    question: "What is the capital of France?",
    options: ["Paris", "London", "Madrid", "Rome"]
  },
  {
    question: "Which planet is known as the Red Planet?",
    options: ["Mars", "Jupiter", "Venus", "Saturn"]
  },
  {
    question: "What is the largest ocean in the world?",
    options: ["Pacific Ocean", "Atlantic Ocean", "Indian Ocean", "Arctic Ocean"]
  },
  {
    question: "Who painted the Mona Lisa?",
    options: ["Leonardo da Vinci", "Pablo Picasso", "Vincent van Gogh", "Michelangelo"]
  },
  {
    question: "What is the chemical symbol for gold?",
    options: ["Au", "Ag", "Fe", "Cu"]
  }
];


export const Assesment = () => {

  const { id, errorMessage } = useAuthStore();
  const { data, loading } = useFetch(`${selectionProcess}/assement/take-exam/1/candidate`)

  const handleSubmit = (event) => {
    event.preventDefault();
    sendAssesment(id);
  };

  return (
    <Container component="main" maxWidth="xs">
      <Box
        sx={{
          marginTop: 8,
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'center',
        }}
      >
        <Typography component="h1" variant="h4">
          Logic Test
        </Typography>
        <Box component="form" onSubmit={handleSubmit} sx={{ mt: 3 }}>
          {
            questions.map((question, index) => (
              <FormControl key={index} component="fieldset" sx={{ mb: 3 }}>
                <FormLabel component="legend">{question.question}</FormLabel>
                <RadioGroup
                  aria-label={`question-${index}`}
                  name={`question-${index}`}
                  defaultValue=""

                >
                  {question.options.map((option, optionIndex) => (
                    <FormControlLabel
                      key={optionIndex}
                      value={option}
                      control={<Radio size="small" />}
                      label={option}
                    />
                  ))}
                </RadioGroup>
              </FormControl>
            ))
          }
          <Grid item sx={{ mt: 2 }}
            xs={12}
            display={errorMessage ? '' : 'none'}
          >
            <Alert severity="error">{errorMessage}</Alert>
          </Grid>
          <Button
            type="submit"
            fullWidth
            variant="contained"
            sx={{ mt: 3, mb: 2 }}
          >
            Send
          </Button>
        </Box>
      </Box>
    </Container>
  );
}
