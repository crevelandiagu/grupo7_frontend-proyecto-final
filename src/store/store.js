import { configureStore } from '@reduxjs/toolkit'
import { authSlice } from './auth'
import { candidateSlice } from './candidate/candidateSlice'

export const store =  configureStore({
  reducer: {
    auth: authSlice.reducer,
    candidate: candidateSlice.reducer,
  },
})