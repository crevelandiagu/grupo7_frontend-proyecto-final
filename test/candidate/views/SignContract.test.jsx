import { fireEvent, render } from '@testing-library/react';
import { SignContract } from '../../../src/candidate/views/SignContract';
import { store } from '../../../src/store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

describe('Test to componenet <SignContract />', () => {
  test('Should be match with snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <SignContract />
        </BrowserRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  })

  test('Call signContract function', () => {

    const signContract = jest.fn();

    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <SignContract signContract={signContract} />
        </BrowserRouter>
      </Provider>
    );

    const form = container.querySelector('form');
    fireEvent.submit(form)

    expect( signContract ).toHaveBeenCalledTimes(0);
  })
});