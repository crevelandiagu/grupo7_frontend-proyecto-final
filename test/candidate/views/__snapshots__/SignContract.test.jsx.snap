// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to componenet <SignContract /> Should be match with snapshot 1`] = `
<div>
  <main
    class="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-binzgt"
    >
      <h1
        class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
      >
        signContract.title
      </h1>
      <form
        class="MuiBox-root css-178yklu"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          readonly=""
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-8ewcdo-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <textarea
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
              disabled=""
              id="contract"
              rows="10"
              style="height: 0px; overflow: hidden;"
            />
            <textarea
              aria-hidden="true"
              class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
              readonly=""
              style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px; width: 100%;"
              tabindex="-1"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  â€‹
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1vddya1-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-1hdyuqk-MuiPaper-root-MuiAlert-root"
            role="alert"
          >
            <div
              class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="ErrorOutlineIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
            </div>
            <div
              class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
            />
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1vhaqj4-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          signContract.signBtn
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </main>
</div>
`;
